### PR TITLE
Toggle modes in execution view

### PR DESCRIPTION
**Pull request by Hemanth-2002**

### CODE REVIEW

File: src/common/components/outline/Outline.tsx [hunk 1]
```diff
@@ -68,14 +77,35 @@ export const Outline: React.FC<Props> = ({
...
 
-      <Flex justify={"flex-end"}>
-        <GButton size="xs" type="secondary" onClick={handleExpandAll}>
-          Expand all
-        </GButton>
-        <GButton size="xs" type="secondary" onClick={handleCollapseAll}>
-          Collapse all
-        </GButton>
-      </Flex>
+      {!expandByDefault && (
```
Comment (pbpranavk) [inline src/common/components/outline/Outline.tsx:80]: should this be a different prop? allowExpand?

File: src/common/components/outline/Outline.tsx [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
   taskSelectionEnabled = false,
   onUpdateOutlineCallback,
+  allowOutlineChanges = false,
+  expandByDefault = false,
 }) => {
   if (!uiOutline) {
     return null;
   }
 
+  if (expandByDefault) {
```
Comment (pbpranavk) [inline src/common/components/outline/Outline.tsx:32]: in this case we will never be able to compress an outline section, therefore we should rename this to alwaysExpand or disableCollapse or something more explicit, what do you think?

File: src/execute/ExecuteView.tsx [hunk 3]
```diff
@@ -164,116 +83,57 @@ export const ExecuteView: React.FC<Props> = ({
     });
   };
 
-  if (isLoading || !exerciseXML) {
-    return <SkeletonScreen />;
-  }
-
-  const blogArticle = new UIBlogArticle(exerciseXML || "");
-  console.log(blogArticle._xml);
+  const renderMode = () => {
```
Comment (pbpranavk) [inline src/execute/ExecuteView.tsx:86]: why is this a function instead of component?

File: src/execute/ExecuteView.tsx [hunk 4]
```diff
@@ -1,29 +1,26 @@
-import React, { useState, useRef, useEffect } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (pbpranavk) [inline src/execute/ExecuteView.tsx:1]: Do we need useEffect?

File: src/execute/ExecuteNotesMode.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      blog_article_xml: string;
+      blog_article_task: string;
+    },
+    context: string,
+    isInitialPrompt: boolean,
+    conversation: { type: string; text: string }[]
+  ) => void;
+};
+
+export const ExecuteNotesMode: React.FC<Props> = ({
```
Comment (pbpranavk) [inline src/execute/ExecuteNotesMode.tsx:29]: What dont we name this NotesMode instead, any specific reason to prepend Execute

File: src/execute/ExecuteOutlineMode.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { UIOutline } from "domain/common/UIOutline";
+import React from "react";
+
+type Props = {
+  exerciseXML: string;
+  growthExerciseId: string | undefined;
+  handleOutlineUpdate: (uiOutline: UIOutline) => void;
+};
+
+export const ExecuteOutlineMode: React.FC<Props> = ({
```
Comment (pbpranavk) [inline src/execute/ExecuteOutlineMode.tsx:13]: What dont we name this OutlineMode instead, any specific reason to prepend Execute

File: src/common/components/outline/Outline.tsx (hunk 1 above)
Comment (Hemanth-2002) [reply src/common/components/outline/Outline.tsx:80]: Done

File: src/common/components/outline/Outline.tsx (hunk 2 above)
Comment (Hemanth-2002) [reply src/common/components/outline/Outline.tsx:32]: Done

File: src/execute/ExecuteView.tsx (hunk 4 above)
Comment (Hemanth-2002) [reply src/execute/ExecuteView.tsx:1]: I think so, We are using it to sync local exerciseXML state with db's exercise.

File: src/execute/ExecuteView.tsx (hunk 3 above)
Comment (Hemanth-2002) [reply src/execute/ExecuteView.tsx:86]: Initially used it as function as all the props to the modes are local to the ExecuteView, Created a component now and sent the required fields as props.

File: src/execute/ExecuteNotesMode.tsx (hunk 5 above)
Comment (Hemanth-2002) [reply src/execute/ExecuteNotesMode.tsx:29]: Done

File: src/execute/ExecuteOutlineMode.tsx (hunk 6 above)
Comment (Hemanth-2002) [reply src/execute/ExecuteOutlineMode.tsx:13]: Done

**Pull request merged**